Footer

mybits opened the pull request:
PT: https://www.pivotaltracker.com/story/show/107913848
design: https://drive.google.com/drive/folders/0B_zI_K-QvNpddHYwY2pOdFVkY1k

mybits committed:
Move footer to partial and add content and style for desktop and tablet version first

mybits committed:
Add mobile version with map marker and phone icons

mybits committed:
Add possibility to call to offices numbers on mobile devices

mybits committed:
Add map links to offices addresses

mybits committed:
Add a newsletter form

mybits committed:
Place addresses tiles even with the other tiles displayed on the site

mybits committed:
Move colors to variables, move blog's social icons to a separate file

tomsabin commented:
Did we decide to not include the company registration number in the end?

tomsabin commented on source/assets/stylesheets/styles/_footer.scss:
```diff
@@ -0,0 +1,159 @@
+  }
+}
+
+.footer-container__registration-address {
```
I'm guessing that this should be displayed somewhere, including the SA registration details..?

tomsabin commented:
The button and input form is rendering incorrectly (rounded borders) on iOS:

![slack for ios upload png](https://cloud.githubusercontent.com/assets/885223/11302520/e9eda13e-8f95-11e5-8e77-3f3abb0862e5.jpeg)

tomsabin commented:
The text input functionality is a bit odd - the placeholder padding is ~30px from the left but when you focus on it, there is no padding.

<img width="370" alt="screen shot 2015-11-20 at 14 53 16" src="https://cloud.githubusercontent.com/assets/885223/11302629/76f591cc-8f96-11e5-964c-6cec582f2e0a.png">
<img width="372" alt="screen shot 2015-11-20 at 14 52 43" src="https://cloud.githubusercontent.com/assets/885223/11302634/7e46bb4a-8f96-11e5-9957-59e8efaaaa66.png">
<img width="382" alt="screen shot 2015-11-20 at 14 53 06" src="https://cloud.githubusercontent.com/assets/885223/11302631/79f3d302-8f96-11e5-89a4-48331a386ad9.png">

tomsabin commented on source/_footer.erb:
```diff
@@ -0,0 +1,67 @@
+     </div>
+    </div>
+    <div class="footer-container__newsletter">
+       <p>Sign up to unboxed updates</p>
```
Capital U? I wonder whether we are "unboxed" or "Unboxed"... I would have thought the latter, but only the logo is lowercase.

mybits replied:
Yes, agreed! Changed it to capital 'U'

tomsabin commented on source/_footer.erb:
```diff
@@ -0,0 +1,67 @@
+          <div class="social-icons__icon--linkedin">
+            <%= image_tag 'linkedin.png', class: 'social-icons__image', alt: 'LinkedIn' %>
+          </div>
+        </a>
```
No more Facebook? :open_mouth:

tomsabin commented:
<img width="844" alt="screen shot 2015-11-20 at 15 23 17" src="https://cloud.githubusercontent.com/assets/885223/11303408/ac536250-8f9a-11e5-88d1-a9722831582d.png">

I'm trying to figure out why the paragraph tag within `footer-container__newsletter` has such a large height. While it might render correctly, I can't help but think something will go wrong here because of it.

It is probably the three floated tiles above, therefore this element might require a clearfix.

tomsabin commented on source/assets/stylesheets/styles/_footer.scss:
```diff
@@ -0,0 +1,159 @@
+  font-weight: 100;
+
+  @include breakpoint($tablet-breakpoint) {
+    padding-left: 16px;
```
Two things here:
1. I think we have quite a mixture of padding dimensions, but we should try and be consistent with them - I've been trying to use 20px left and right as a minimum after realising this (I have also update the project story styles on the `homepage` branch). Currently, on those pages this footer fits nicely with the related content above (which is also 16px), but on other pages such as the blog, the padding is 20px.
2. Secondly is because we only have a left padding, declaring 50% for the newsletter is showing larger than 50% (maybe 51/52%) because there is no padding on the right. Because of that the newsletter looks ever-so-slightly off. Hopefully the screenshot will be able to explain.

<img width="843" alt="screen shot 2015-11-20 at 15 28 33" src="https://cloud.githubusercontent.com/assets/885223/11303685/063975c4-8f9c-11e5-8aec-fa1fba625bb2.png">

mybits replied:
Oh yes, that was actually changed in later commits, but now since we've changed the layout I guess it'll need another review :)

tomsabin commented on source/assets/stylesheets/styles/_footer_social_icons.scss:
```diff
@@ -1,5 +1,4 @@
-  background-color: $footer-social-icons__background-color;
   text-align: right;
 }
 
```
Is this file needed anymore?

mybits replied:
Good spot! Removed

tomsabin commented on source/_footer.erb:
```diff
@@ -0,0 +1,67 @@
+      <p class="footer-container__company-name">
+        unboxed
+      </p>
+      <p>hello@unboxed.co</p>
```
Could this be a `mailto`?

mybits replied:
Done!

tomsabin commented on source/_footer.erb:
```diff
@@ -0,0 +1,67 @@
+      </div>
+      <div class="footer-container__registration-address">
+        <p>registered in England and Wales</p>
+        <p>05440833</p>
```
_Maybe_ we could link to https://beta.companieshouse.gov.uk/company/05440833 :stuck_out_tongue:

mybits replied:
I'll ask Graeme about it :)

mybits replied:
He said, not necessary :) so I'll leave it as it is

tomsabin commented on source/_footer.erb:
```diff
@@ -0,0 +1,67 @@
+      <p class="footer-container__address-city"><a href="http://goo.gl/maps/AuKrs" target="_blank"><i class="fa fa-map-marker footer__map-icon"></i>
+        London
+      <span>office</span></a></p>
+      <a class="footer-container__address-phone" href="tel:+442071834250"><i class="fa fa-phone footer__phone-icon"></i>
```
:+1:

tomsabin commented on source/_footer.erb:
```diff
@@ -0,0 +1,67 @@
+        <p>17 Blossom Street</p>
+        <p>London, E1 6PL</p>
+        <p>United Kingdom</p>
+      </div>
```
Similar to the mobile view, we could link the address to Google maps

mybits replied:
We got it in London/Cape Town office - guess that would be enough?

mybits committed:
Add registrations addresses, increase company name font

mybits committed:
Change layout and move newsletter to be in the first tile instead in the bottom of the footer,
move social icons to the bottom central, remove company name and leave only the email address

mybits committed:
Remove unused style file and variable

mybits committed:
Add more spacing to an email input in the newsletter form

tomsabin commented on source/_footer.erb:
```diff
@@ -0,0 +1,76 @@
+          <p>17 Blossom Street</p>
+          <p>London, E1 6PL</p>
+          <p>United Kingdom</p>
+        </div>
```
It has just occurred to me that we could probably use an [`address`](http://www.w3schools.com/tags/tag_address.asp) tag here.

> The address element provides contact information for a document or part of a document. Information provided by address may include the names of the document’s maintainers, links to the maintainers’ Web pages, e-mail addresses for feedback, postal addresses, phone numbers, and so on. The address element is not appropriate for all postal and e-mail addresses; it should be reserved for providing such information about the contact people for the document.

mybits committed:
Fix newsletter form on iOS devices

mybits committed:
Add more space to social icons bar

mybits commented:
@tomsabin thanks for all the comments, all have been addressed in separate commits

mybits committed:
Use address tag where relevant

mybits committed:
Make newsletter form transition between tablet and desktop to be more smooth,
place social icons slightly nearer to the top on mobile

mybits committed:
Cleaning newsletter forms style a bit, adding hover color to submit button,
align office addresses tiles with the page tiles content

mybits committed:
Move phone numbers below addresses, remove countries,
add map-marker icons on tablet and desktop, plus minor style improvements

mybits committed:
Fix font weight on newsletter form

mybits committed:
Copy change in a newsletter

mybits committed:
Change cursor on subscribe button to a pointer, fix uneven height on newsletter form

tomsabin commented:
LGTM :v:
